packages/integration/src/azure/core.ts

```diff
@@ -100,7 +100,17 @@ export function getAzureDownloadUrl(url: string): string {
     ? `&scopePath=${encodeURIComponent(filepath)}`
     : '';
 
-  return `${protocol}://${resource}/${organization}/${project}/_apis/git/repositories/${repoName}/items?recursionLevel=full&download=true&api-version=6.0${scopePath}`;
+  if (url.includes('dev.azure.com')) {
+    return `${protocol}://${resource}/${organization}/${project}/_apis/git/repositories/${repoName}/items?recursionLevel=full&download=true&api-version=6.0${scopePath}`;
+  }
+
+  // For Azure DevOps Server `parseGitUrl` returns the same values
+  // for `organization` and `project` like this: `organization/project/_git`
```

Comment:
Oh wow, this really looks like a bug in the git-url-parse library, sounds like it might even be trying to interpret it as the wrong provider or something like that.

The tests don't cover any non-cloud URLs which is slightly concerning, maybe we should be looking at contributing there some day...

Reply:
Yes, I tried to see if I could make this change upstream in `git-url-parse` but felt like I was opening up a large can of worms. Doing it this way makes the surface area of the change much smaller.

I did try to add some tests for this to slowly get some non-cloud URLs in the test suite :)